runtime/onert/core/src/ir/train/UseDefGenerator.test.cc

```diff
@@ -0,0 +1,2027 @@
+                        expected_forwarding_ba_input1_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input2_uses{
+      forwarding_ba_index, backwarding_ba_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input2_def{};
+    const auto expected_forwarding_ba_input2 =
+      createUseDefChain(tgraph.operands().at(ba_input2), expected_forwarding_ba_input2_uses,
+                        expected_forwarding_ba_input2_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_pred_uses{
+      forwarding_loss_index, backwarding_loss_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_pred_def{forwarding_ba_index};
+    const auto expected_forwarding_y_pred =
+      createUseDefChain(tgraph.operands().at(y_pred), expected_forwarding_y_pred_uses,
+                        expected_forwarding_y_pred_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_true_uses{
+      forwarding_loss_index, backwarding_loss_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_true_def{};
+    const auto expected_forwarding_y_true =
+      createUseDefChain(tgraph.operands().at(y_true), expected_forwarding_y_true_uses,
+                        expected_forwarding_y_true_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_output_uses{};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_output_def{
+      forwarding_loss_index};
+    const auto expected_forwarding_output =
+      createUseDefChain(tgraph.operands().at(output), expected_forwarding_output_uses,
+                        expected_forwarding_output_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_input_uses{};
+    std::vector<train::TrainingOperationIndex> expected_backwarding_input_def{backwarding_fc_index};
+    const auto expected_backwarding_input = createUseDefChain(
+      tgraph.operands().at(input), expected_backwarding_input_uses, expected_backwarding_input_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_weights_uses{};
+    std::vector<train::TrainingOperationIndex> expected_backwarding_weights_def{
+      backwarding_fc_index};
+    const auto expected_backwarding_weights =
+      createUseDefChain(tgraph.operands().at(weights), expected_backwarding_weights_uses,
+                        expected_backwarding_weights_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_ba_input1_uses{
+      backwarding_fc_index};
+    std::vector<train::TrainingOperationIndex> expected_backwarding_ba_input1_def{
+      backwarding_ba_index};
+    const auto expected_backwarding_ba_input1 =
+      createUseDefChain(tgraph.operands().at(ba_input1), expected_backwarding_ba_input1_uses,
+                        expected_backwarding_ba_input1_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_ba_input2_uses{};
```

Comment:
The "uses" of the second back-prop tensor of backwarding "add" node is empty because it is a constant in forwarding and never be back-propagated.